packages/react-interactive-tab/src/components/InteractiveTab/InteractiveTab.test.tsx

```diff
@@ -0,0 +1,122 @@
+import type { TabListContextValue } from '@fluentui/react-components';
+import { CalendarMonthRegular } from '@fluentui/react-icons';
+import { InteractiveTab } from './InteractiveTab';
+
+describe('InteractiveTab', () => {
+  const defaultContext: TabListContextValue = {
+    appearance: 'transparent',
+    disabled: false,
+    size: 'medium',
+    vertical: false,
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onRegister: () => {},
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onUnregister: () => {},
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onSelect: () => {},
+    getRegisteredTabs: () => {
+      return {
+        registeredTabs: {},
+      };
+    },
+  };
+
+  it('renders correctly', () => {
+    const contextValues = {
+      tabList: { ...defaultContext },
+    };
+
+    const result = render(
+      <TabListProvider value={contextValues.tabList}>
+        <InteractiveTab value="1" contentBefore="Before" contentAfter="After">
+          Default Tab
+        </InteractiveTab>
+      </TabListProvider>
+    );
+
+    expect(result.container).toMatchSnapshot();
+  });
+
+  it('selected when clicked', () => {
+    const onSelect = jest.fn();
+
+    const contextValues = {
+      tabList: { ...defaultContext, onSelect },
+    };
+
+    const result = render(
+      <TabListProvider value={contextValues.tabList}>
+        <InteractiveTab value="1" contentBefore="Before" contentAfter="After">
+          Default Tab
+        </InteractiveTab>
+      </TabListProvider>
+    );
+
+    fireEvent.click(result.getByRole('tab'));
+    expect(onSelect).toHaveBeenCalledWith(expect.anything(), { value: '1' });
+  });
+
+  it.each([
+    ['default', { ...defaultContext }],
+    ['subtle appearance', { ...defaultContext, appearance: 'subtle' }],
+    ['vertical', { ...defaultContext, vertical: true }],
+    ['small size', { ...defaultContext, size: 'small' }],
+    [
+      'small size and vertical',
+      { ...defaultContext, size: 'small', vertical: true },
+    ],
+    ['medium size', { ...defaultContext, size: 'medium' }],
+    [
+      'medium size and vertical',
+      { ...defaultContext, size: 'medium', vertical: true },
+    ],
+    ['large size', { ...defaultContext, size: 'large' }],
+    [
+      'large size and vertical',
+      { ...defaultContext, size: 'large', vertical: true },
+    ],
+  ])('renders %s correctly with icon slotted', (_testName, tabList) => {
+    const contextValues = {
+      tabList: tabList as TabListContextValue,
+    };
+
+    const result = render(
+      <TabListProvider value={contextValues.tabList}>
+        <InteractiveTab
+          icon={<CalendarMonthRegular />}
+          value="1"
+          contentBefore="Before"
+          contentAfter="After"
+        >
+          Default Tab
+        </InteractiveTab>
+      </TabListProvider>
+    );
+
+    expect(result.container).toMatchSnapshot();
```

Comment:
I'm not sure what's the value of dumping these huge html structures in terms of testing quality and guaranteing that things work.

we should probably either use browser test ( via playwright ) or narrow down only to particular parts you wanna assert on.

Reply:
I wanted to use the same tests as for the core library's Tab component to ensure consistency between the extension and the core component.

Reply:
yeah, so what happens in core is not the best approach on how to write test per say, but there were some historicall reason and another things that didn't make it "to stable/official guidance" 

when we have opportunity to not follow set antipatterns lets try to do that